src/ray/util/macros.h

```diff
@@ -30,7 +30,7 @@
 // the absence of better information (ie. -fprofile-arcs).
 //
 #if defined(__GNUC__)
-#define RAY_PREDICT_FALSE(x) (__builtin_expect(x, 0))
+#define RAY_PREDICT_FALSE(x) (__builtin_expect((x), 0))
```

Comment:
macro argument should wrap with `()`, for example, `,` itself is a valid operator.

Reply:
do we also need `!!` ?

Reply:
No, I checked abseil implementation: 
https://github.com/abseil/abseil-cpp/blob/feb6aab8db3315cc375351ced7ca0ea4db5b4599/absl/base/optimization.h#L187-L194

Reply:
I could remove and see if anything breaks, my groundless guess is: the author didn't intend to allow non-boolean types, so he/she perform negate operation which only works for boolean.

Reply:
Update:
- Proved to be necessary, otherwise `RAY_DCHECK(uint8_t*)` doesn't compile
- My guess is: it makes a type conversion from pointer to boolean

Reply:
Well, I prefer to have an explicit function which takes a boolean argument, checkout https://github.com/ray-project/ray/pull/48663/commits/e29c98a9e640f6866e2dbae4d4b292e20c0b82ef

Reply:
do we want to follow absl to use 

#define RAY_PREDICT_FALSE(x) (__builtin_expect(false || (x), false))
#define RAY_PREDICT_TRUE(x) (__builtin_expect(false || (x), true))

?

Reply:
Yeah sure, second thought, abseil's `false || ` is also used to convert whatever-type-x-is to boolean.

Reply:
Implementation updated to abseil's implementation.